### PR TITLE
docs: add Test Starter

### PR DESCRIPTION
**Pull request by maxreichmann**
Dev Guide explaining Test Starter: https://ui5.sap.com/#/topic/032be2cb2e1d4115af20862673bedcdb

JIRA: CPOUI5FOUNDATION-819

### CODE REVIEW

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:46]: Why do I need to add the "namespace" to the formatter path? 
I think this should be explained here as well so learners can extrapolate from here to their use case.

Is this related to the fact that a new namespace is defined in Test.qunit.html?
If so it mightbe easier then to first create Test.qunit.html before creating unit/model/formatter.ts?

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:97]: At least for me as a beginner, this is not enough to understand the implementation below: Why script id)"sap-ui-bootstrap" is needed here? What is "../resources/sap/ui/test/starter/createSuite.js" for? Why do I need a new namespace "test-resources.ui5.walkthrough"? Does it need to be test-resources+namespace of my initial app or could this be any name?
What is the configuration option "data-sap-ui-testsuite" for and how to define it?
Was does these two divs at the end to? To which id="qunit" and id="qunit-fixture" are referencing?
...?

It seems as it would make sene to create this file only after testsuite.qunit.ts was created, so learners understand on which this value is based.

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:122]: Not sure a beginner will understand this configuration. Are all of these parameters mandatory for each test? What are they doing and why are they needed? How do I know about the available configuration options? 
...?
I'm also not sure I understand what this configuration does (why it is needed)?
I believe to understand that it does something with file test-resources/ui5/walkthrough/Test.qunit.html. But at this point of the tutorial this file does not exist yet. Maybe it would make sense to first create Test.qunit.html and only then create (and explain) testsuite.qunit.ts?

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:164]: I guess it would be good to at least explain src="../resources/sap/ui/test/starter/runTest.js"

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:161]: Is it indeed best practice (or even mandatory) to start the name of this file with an upper "T"? Just wondering as all other html files start with lower case. If so this should be mentioned. If specific naming convention exist, this should be also mentioned in the Convention section.

**Review by margopolo — COMMENTED** [review]
Are there indeed no additional resources (e.g. chapters in the documentation) about the test starter?

In general, I believe there is still some explanation needed so beginners can understand and extrapolate from the tutorial to their use cases.

In addition, the link for previewing this step on the github page is not adapted also file test/unit/unitTests-cdn.qunit.html was deleted. I would assume that two additional files (testsuite-cdn.qunit.html and Test-cdn.qunit.hmtl) are still needed to compliment this step.
Also, in the readme we would need a new image (or rather link) for the folder structure, as the current one does not fit anymore.

**Comment by margopolo** [comment]
What is the minimum OpenUI5 version required for the test starter?
We currently require version 1.129. If a higher version is necessary, all steps in this tutorial will need to be updated to match the new version.

**Comment by maxreichmann** [comment]
> Are there indeed no additional resources (e.g. chapters in the documentation) about the test starter?
> 
> In general, I believe there is still some explanation needed so beginners can understand and extrapolate from the tutorial to their use cases.
> 
> In addition, the link for previewing this step on the github page is not adapted also file test/unit/unitTests-cdn.qunit.html was deleted. I would assume that two additional files (testsuite-cdn.qunit.html and Test-cdn.qunit.hmtl) are still needed to compliment this step. Also, in the readme we would need a new image (or rather link) for the folder structure, as the current one does not fit anymore.

Regarding additional resources, [these Dev Guide chapters](https://ui5.sap.com/#/topic/032be2cb2e1d4115af20862673bedcdb) could be helpful.

Regarding the link for previewing this step on the github page: I don't fully understand. How exactly does this preview work? The `README.me` file can at least be viewed as the compiled markdown code locally.

Regarding the deletion of the two cdn files. I assumed these two files are not necessary anymore with the introduction of the Test Starter.

A new image of the new folder structure is indeed necessary and will be added by UA colleagues.

File: steps/27/README.md
Comment (codeworrior) [reply steps/27/README.md:161]: Naming of generic page for apps: there is neither a technical restriction nor a best practice in any direction. The original Test.qunit.html in openui5 used "T" because most test pages in UI5 use capitalized CamelCase. But some teams who defined their own generic test page, already use a lower case name.

If you feel that a lower case name better fits in the context of apps, I see no reason not to convert this to lowercase.

**Comment by codeworrior** [comment]
@margopolo AFAIK, minimal version for using it in apps is 1.124.

**Comment by matz3** [comment]
Now it's ready for review. We'll update the other steps once the content is approved to keep the changes for review as small as possible. Only steps 26 and 27 make changes in the `test` folder, so it can just be overwritten.

File: steps/27/webapp/test/Test.cdn.qunit.html
Comment (flovogt) [inline steps/27/webapp/test/Test.cdn.qunit.html:6]:
```suggestion
		src="https://sdk.openui5.org/1.130.0/resources/sap/ui/test/starter/runTest.js"
```
Should we use the latest published UI5 version?

Comment (matz3) [reply steps/27/webapp/test/Test.cdn.qunit.html:6]: The whole tutorial currently uses 1.129, so I didn't want to change that in this PR. An update to 1.130 makes sense to me but I'd rather do it in a separate PR.

Comment (flovogt) [reply steps/27/webapp/test/Test.cdn.qunit.html:6]: Thx a lot for clarification.

**Review by flovogt — APPROVED** [review]
@LilyanaOviPe could please have a look

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:88]: Somehow the explanation about the implementation in the file itself got lost. How about to change the description to:

This module will serve as the entry point for all our unit tests. It will be referenced in the test suite that we will set up later on.

Inside the  `unitTests.qunit.ts` file, we import the unit test for the custom formatter. This ensures that any tests related to the custom formatter functionality will be included when running our unit tests.

Comment (matz3) [reply steps/27/README.md]: Thanks 👍🏻 I've update it as suggested.

File: steps/27/README.md
Comment (matz3) [reply steps/27/README.md:155]: Done

File: steps/27/README.md
Comment (margopolo) [inline steps/27/README.md:101]: Can we have here one additional sentence about the src="../resources/sap/ui/test/starter/runTest.js" script that is initialized in the script tag? Would be good to also mention what this script is for.

Comment (matz3) [reply steps/27/README.md:101]: Done

**Review by margopolo — DISMISSED** [review]

**Comment by matz3** [comment]
Please do not merge yet. I will now update the files in all following steps.

**Comment by matz3** [comment]
Ready for final review

**Review by margopolo — APPROVED** [review]

File: steps/28/README.md
Comment (LilyanaOviPe) [inline steps/28/README.md:161]: If we now open the `webapp/test/testsuite.qunit.html` file in the browser and select `integration/opaTests`, the QUnit layout should appear and a test “Should see the Hello dialog” will run immediately. This action will load the app component on the right side of the page. There you can see the operations the test is performing on the app. If everything works correctly, a button click will be triggered, then a dialog will be displayed and the test case will be green.

File: steps/27/README.md
Comment (matz3) [reply steps/27/README.md:44]: Done via https://github.com/SAP-samples/ui5-typescript-walkthrough/pull/77

**Pull request merged**
